### PR TITLE
framework: Add sead::Task header

### PR DESCRIPTION
**Pull request by Sanae6**
Added a header for sead::Task.

- - -
This change is [Reviewable](https://reviewable.io/reviews/open-ead/sead/130)

### CODE REVIEW

**Comment by aboood40091** [comment]
I have never heard of sead::Task. Do you have a source for that?

**Comment by Sanae6** [comment]
@aboood40091 I found sead::Task in SMO.
RootTask, a critical part of early startup code, depends on it.
javaw_zm6SO1493Z

File: include/framework/seadTask.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    void attachCalcImpl() override;
+    void attachDrawImpl() override;
+    void detachCalcImpl() override;
+    void detachDrawImpl() override;
+    const RuntimeTypeInfo::Interface* getCorrespondingMethodTreeMgrTypeInfo() const override;
+    MethodTreeNode* getMethodTreeNode(s32 method_type) override;
+    virtual void calc() {}
+    virtual void draw() {}
+
+protected:
```
Comment (MonsterDruide1) [inline include/framework/seadTask.h:32]: Why would you want to make this `protected` instead of `private`?

Comment (Sanae6) [reply include/framework/seadTask.h:32]: I used [seadCalculateTask.h](https://github.com/open-ead/sead/blob/d42a8864af9eab6c869820836fa15d47992202aa/include/framework/seadCalculateTask.h) as a base, and only slightly changed what was in that file. My assumption was that it was protected so the method tree nodes could be accessible to its children, but I didn't give it much thought past that.

Comment (MonsterDruide1) [reply include/framework/seadTask.h:32]: Alright, makes sense though.

**Pull request merged**